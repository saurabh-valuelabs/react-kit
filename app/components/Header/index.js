/**
 *
 * Navigation
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { AppSidebarToggler } from '@coreui/react';

function Header({ imgSrc, altText }) {
  return (
    <>
      <AppSidebarToggler className="d-lg-none" display="md" mobile />
      <a className="navbar-brand" href="#/">
        <img src={imgSrc} alt={altText} className="logo-img" />
      </a>
    </>
  );
}

Header.propTypes = {
  imgSrc: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
};

export default Header;
