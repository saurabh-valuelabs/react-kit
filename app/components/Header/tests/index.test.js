/**
 *
 * Tests for Header
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import nav from '../../../config/_nav';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Header from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

jest.mock('react-router-dom', () => ({ Link: 'a' }));
describe('<Header />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Header nav={nav} imgSrc="" altText="" />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Header nav={nav} imgSrc="" altText="" />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
