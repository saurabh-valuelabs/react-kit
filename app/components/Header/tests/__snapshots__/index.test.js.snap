// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render and match the snapshot 1`] = `
<header>
  <nav
    class="row px-0 navbar navbar-expand-lg navbar"
  >
    <div
      class="col-6 col-lg-auto"
    >
      <a
        class="navbar-brand"
        to="/"
      >
        <img
          alt=""
          src=""
        />
      </a>
    </div>
    <div
      class="lg-hide col-12"
    >
      <button
        aria-hidden="true"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
    </div>
    <div
      class="col"
    >
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              title="Dashboard"
              to="/"
            >
              Dashboard
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              title="Funds Listing"
              to="/fund-listing"
            >
              Funds Listing
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              title="My Profile"
              to="/my-profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <hr />
</header>
`;
